src/ir/omake_cache.ml

```diff
@@ -779,6 +779,12 @@ let has_file_stat ?compact_stat ?digest cache node =
     has_file_stat_1 ?compact_stat ?digest cache node
 
 
+let unix_filename_exists path =
+  let basedir = Filename.dirname path in
+  let basename = Filename.basename path in
+  basename = "." || basename = ".." || Array.exists (String.equal basename) (Sys.readdir basedir)
```

Comment:
readdir inside stat (wrapper) sounds expensive

Reply:
It is :-(, but I'm not aware of any BSD/POSIX syscall/API to obtain the original capitalization -- I read even `realpath(3)` lies. And I don't know how to determine whether the filesystem is case-insensitive to restrict the above check to that case (without write ops that is). At least it's performed only when there's a successful (l)stat.
The only